src/main/java/org/jabref/gui/maintable/MainTableHeaderContextMenu.java

```diff
@@ -66,15 +82,20 @@ private void constructItems(MainTable mainTable) {
             }
         }
 
-        SeparatorMenuItem separator = new SeparatorMenuItem();
-        this.getItems().add(separator);
+        if (!commonColumns.isEmpty()) {
+            this.getItems().add(new SeparatorMenuItem());
 
-        // Append to the menu the current remaining columns in the common columns.
-
-        for (TableColumn<BibEntryTableViewModel, ?> tableColumn : commonColumns) {
-            RightClickMenuItem itemToAdd = createMenuItem(tableColumn, false);
-            this.getItems().add(itemToAdd);
+            // Append to the menu the current remaining columns in the common columns.
+            for (TableColumn<BibEntryTableViewModel, ?> tableColumn : commonColumns) {
+                RightClickMenuItem itemToAdd = createMenuItem(tableColumn, false);
+                this.getItems().add(itemToAdd);
+            }
         }
+
+        this.getItems().add(new SeparatorMenuItem());
+        ActionFactory actionfactory = new ActionFactory(Globals.getKeyPrefs());
+        MenuItem showMoreItem = actionfactory.createMenuItem(RightClickMenuActions.SHOW_PREFS_RIGHT_CLICK_MENU, new ShowPreferencesAction(frame, taskExecutor, "Entry table"));
```

Comment:
Please use the class instead of a magic string.

Reply:
You probably misunderstood me. Use TableTab.class, just the class object, not an instance of tablets.

Reply:
I tried to resolve it. I'm not sure I did the way intended

Reply:
Yes, thats it what I meant. This way, if the code is being refactored in the future by an ide the names are automatically changed too. Reduces maintenance.